src/main/java/frc/trigon/robot/subsystems/arm/simulationarm/SimulationArmIO.java

```diff
@@ -0,0 +1,94 @@
+
+    @Override
+    protected void updateInputs(ArmInputsAutoLogged inputs) {
+        angleMotor.update(RobotConstants.PERIODIC_TIME_SECONDS);
+        elevatorMotor.update(RobotConstants.PERIODIC_TIME_SECONDS);
+
+        inputs.angleMotorVoltage = angleVoltage;
+        inputs.angleMotorCurrent = angleMotor.getCurrentDrawAmps();
+        inputs.angleVelocityDegreesPerSecond = getAngleVelocityDegreesPerSecond();
+        inputs.anglePositionDegrees = Units.radiansToDegrees(angleMotor.getAngleRads());
+
+        inputs.elevatorMotorVoltage = elevatorVoltage;
+        inputs.elevatorMotorCurrent = elevatorMotor.getCurrentDrawAmps();
+        inputs.elevatorPositionRevolution = getElevatorPositionRevolution();
+        inputs.elevatorVelocityRevolutionsPerSecond = elevatorMotor.getVelocityMetersPerSecond() * SimulationArmIOConstants.METERS_PER_REVOLUTION;
+    }
+
+    @Override
+    protected void stopAngleMotors() {
+        setAngleVoltage(0);
+    }
+
+    @Override
+    protected void stopElevatorMotors() {
+        setElevatorVoltage(0);
+    }
+
+    @Override
+    protected void setTargetAngleState(TrapezoidProfile.State targetState) {
+        setAngleVoltage(calculateAngleOutput(targetState));
+    }
+
+    @Override
+    protected void setTargetElevatorState(TrapezoidProfile.State targetState) {
+        setElevatorVoltage(calculateElevatorOutput(targetState));
+    }
+
+    private void setAngleVoltage(double voltage) {
+        this.angleVoltage = voltage;
+        angleMotor.setInputVoltage(voltage);
+    }
+
+    private void setElevatorVoltage(double voltage) {
+        this.elevatorVoltage = voltage;
+        elevatorMotor.setInputVoltage(voltage);
+    }
+
+    private double calculateAngleOutput(TrapezoidProfile.State targetState) {
+        double pidOutput = SimulationArmIOConstants.ANGLE_PID_CONTROLLER.calculate(
+                angleMotor.getAngleRads(),
+                targetState.position
+        );
+        double feedforward = SimulationArmIOConstants.ANGLE_FEEDFORWARD.calculate(
+                Units.degreesToRadians(targetState.position),
+                Units.degreesToRadians(targetState.velocity)
+        );
+        return pidOutput + feedforward;
+    }
+
+    private double calculateElevatorOutput(TrapezoidProfile.State targetState) {
+        double pidOutput = SimulationArmIOConstants.ELEVATOR_PID_CONTROLLER.calculate(
+                elevatorMotor.getPositionMeters(),
+                targetState.position
+        );
+        double feedforward = SimulationArmIOConstants.ELEVATOR_FEEDFORWARD.calculate(targetState.velocity);
+        return pidOutput + feedforward;
+    }
+
+    private double getAngleVelocityDegreesPerSecond() {
+        return Units.radiansToDegrees(angleMotor.getAngleRads());
+    }
+
+    private double getElevatorPositionRevolution() {
+        if (elevatorMotor.getPositionMeters() == SimulationArmIOConstants.MIN_ELEVATOR_HEIGHT_METERS)
+            return 0;
```

Comment:
This is not what I meant... I meant the sims point of reference is wrong, so you should fix it. Kinda like offset.

Reply:
Wdym "sims point of reference"?

Reply:
When the sim is all the way down, its "encoder" will tell us its position is the min position. We would want this to be 0 rather than the min position. But also, change this for everything. So sort of offset it by the min position so it'll give us the point of reference we want.

Reply:
I don't understand. what offset do I need to add?

Reply:
what do u want me to do?